File: js/src/Stack.js
/** The Stack Kata **/

class Stack {
  constructor() {
    this.content = [];
  }

  getSize() {
    return this.content.length;
  }

  isEmpty() {
    return 0 === this.getSize();
  }

  push( variable ) {
    this.content[this.content.length] = variable;
  }

  pop() {
    if ( this.isEmpty() ) {
      throw new Error("StackUnderflow");
    }

    return this.content.splice(this.content.length - 1, 1)[0];
  }

  top() {
    if (this.isEmpty()){
      throw new Error("Empty");
    }
    return this.content[this.content.length - 1];
  }

  find(value){
    for( var i = this.getSize() - 1; i >= 0; i-- ){
      if( value ===  this.content[i]){
        return (this.getSize()-1)-i;
      }
    }
    return null;
  }
}

File: Stack/python/Stack.py
#!/usr/local/bin/python3
class Stack:
	def __init__(self):
		self.size = 0
		self.values = []

	def isEmpty(self):
		return 0 == self.getSize()

	def push(self, value):
		self.values.insert(0, value)

	def getSize(self):
		return len(self.values)

	def pop(self):
		if self.isEmpty():
			raise self.Underflow("Underflow")
		return self.values.pop(0)

	def top(self):
		if self.isEmpty():
			raise self.Empty("Empty")
		return self.values[0]

	def find(self, value):
		try:
			return self.values.index(value)
		except:
			return None

	class Underflow(Exception):
		pass

	class Empty(Exception):
		pass
